ChaosMod/Effects/db/Player/PlayerTpController.cpp

```diff
@@ -157,8 +157,10 @@ static void OnStartFront()
 {
 	Ped playerPed = PLAYER_PED_ID();
 	Vector3 newPos = GET_OFFSET_FROM_ENTITY_IN_WORLD_COORDS(playerPed, 0.f, 50.f, 0.f);
+	float groundZ;
+	GET_GROUND_Z_FOR_3D_COORD(newPos.x, newPos.y, 1000.f, &groundZ, false, false);
```

Comment:
you should check, if it succeeded and use "newPos.z" as a fallback

Reply:
Added the check!

Reply:
👌